### PR TITLE
M/layer settings

### PR DESCRIPTION
**Pull request by mfelsche**

### CODE REVIEW

File: src/crate/testing/layer.py [hunk 1]
```diff
@@ -40,34 +42,52 @@ def __init__(self,
...
-            '-Des.discovery.type=zen',
-            '-Des.discovery.zen.ping.multicast.enabled=%s' % ("true" if multicast else "false"),
-            '-Des.config=%s' % crate_config,
-            '-Des.path.conf=%s' % os.path.dirname(crate_config),
-        )
-        if transport_port:
-            start_cmd += ('-Des.transport.tcp.port=%s' % transport_port,)
-        super(CrateLayer, self).__init__(name, servers=servers, start_cmd=start_cmd)
+        settings = self.create_settings(crate_config, cluster_name, name, host, port, transport_port, multicast, settings)
+        start_cmd = (crate_exec, ) + tuple(["-Des.%s=%s" % (setting, value) for setting, value in settings.items()])
```
Comment (chaudum) [inline src/crate/testing/layer.py]: `settings.items()` returns a tuple that can be used for string formatting directly
e.g. `start_cmd = (crate_exec, ) + tuple(["-Des.%s=%s" % opt for opt in settings.items()])`

File: tox.ini [hunk 2]
```diff
@@ -23,7 +23,7 @@ deps =
     zope.testing
     zc.customdoctests
     lovely.testlayers
-    sqlalchemy>=0.9.1
+    sqlalchemy>=0.9.1,<1.0.0
```
Comment (chaudum) [inline tox.ini]: can we fix that in the `versions.cfg` too?

**Comment by chaudum** [comment]
ack

File: tox.ini (hunk 2 above)
Comment (mfussenegger) [inline tox.ini]: what about setup.py?

**Pull request merged**